### DISCUSSION TITLE
ValueError: zero-size array to reduction operation minimum which has no identity.

### DISCUSSION THREAD

**Discussion by AIfanss** [MIC-DKFZ/nnUNet#1327] [Q&A]
hi, when I run the nnUNet_plan_and_preprocess.py, ValueError: zero-size array to reduction operation minimum which has no identity appeared, it happened when I run it on my own dataset while on the given official dataset everything was fine. So there has to be something wrong with my own dataset, but I can tell the difference, could someone help me with this ?

**Comment by FabianIsensee** [comment]
Have you tried `nnUNet_plan_and_preprocess [...] --verify_dataset_integrity`?

**Reply by billhhh** [reply]
Tried, returns the same issue

**Comment by OceaneerDs** [comment]
Hi there!
@AIfanss I probably ran into the same error with _nnUNetv2_plan_and_preprocess -d 40 --verify_dataset_integrity_ . Did you solve yours?
My error source seems to be: 
_UNet\nnUNet\nnunetv2\experiment_planning\dataset_fingerprint\fingerprint_extractor.py
--> min': float(np.min(foreground_intensities_per_channel[i]))_ 

**foreground_intensities_per_channel[i] is of size zero.**

### Output of _nnUNetv2_plan_and_preprocess -d 40 --verify_dataset_integrity_ :
_Fingerprint extraction...
Dataset040_Voids
Using <class 'nnunetv2.imageio.natural_image_reager_writer.NaturalImage2DIO'> as reader/writer

####################
verify_dataset_integrity Done.
If you didn't see any error messages then your dataset is most likely OK!
####################

Using <class 'nnunetv2.imageio.natural_image_reager_writer.NaturalImage2DIO'> as reader/writer_
...\Desktop\Projects\UNet\unet_venv\lib\site-packages\numpy\core\fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
...\Desktop\Projects\UNet\unet_venv\lib\site-packages\numpy\core\_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
...\Desktop\Projects\UNet\unet_venv\lib\site-packages\numpy\core\_methods.py:206: RuntimeWarning: Degrees of freedom <= 0 for slice
  ret = _var(a, axis=axis, dtype=dtype, out=out, ddof=ddof,
...\Desktop\Projects\UNet\unet_venv\lib\site-packages\numpy\core\_methods.py:163: RuntimeWarning: invalid value encountered in divide
  arrmean = um.true_divide(arrmean, div, out=arrmean,
...\Desktop\Projects\UNet\unet_venv\lib\site-packages\numpy\core\_methods.py:198: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File ..\Desktop\Projects\UNet\unet_venv\Scripts\nnUNetv2_plan_and_preprocess.exe\__main__.py", line 7, in <module>
  File ...\Desktop\Projects\UNet\nnUNet\nnunetv2\experiment_planning\plan_and_preprocess_entrypoints.py", line 182, in plan_and_preprocess_entry
    extract_fingerprints(args.d, args.fpe, args.npfp, args.verify_dataset_integrity, args.clean, args.verbose)
  File ...\Desktop\Projects\UNet\nnUNet\nnunetv2\experiment_planning\plan_and_preprocess_api.py", line 47, in extract_fingerprints
    extract_fingerprint_dataset(d, fingerprint_extractor_class, num_processes, check_dataset_integrity, clean,
  File ...\Desktop\Projects\UNet\nnUNet\nnunetv2\experiment_planning\plan_and_preprocess_api.py", line 33, in extract_fingerprint_dataset
    return fpe.run(overwrite_existing=clean)
  File ...\Desktop\Projects\UNet\nnUNet\nnunetv2\experiment_planning\dataset_fingerprint\fingerprint_extractor.py", line 185, in run
    'min': float(np.min(foreground_intensities_per_channel[i])),
  File ...\Desktop\Projects\UNet\unet_venv\lib\site-packages\numpy\core\fromnumeric.py", line 2953, in min
    return _wrapreduction(a, np.minimum, 'min', axis, None, out,
  File ...\Desktop\Projects\UNet\unet_venv\lib\site-packages\numpy\core\fromnumeric.py", line 88, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
**ValueError: zero-size array to reduction operation minimum which has no identity****

**Reply by adriroyu** [reply]
Hi! I ran into the same error. Did you managed to solve it?

**Reply by OceaneerDs** [reply]
Hi @adriroyu!
No, I stopped trying after a short time and implemented my own solution.

**Comment by billhhh** [comment]
Same issue here

**Reply by billhhh** [reply]
Solved, thanks, for me I wrongly called copy_BraTS_segmentation_and_convert_labels_to_nnUNet() function, as I modify my script based on brats one. Direct copy of masks solved this issue

**Comment by Flock1** [comment]
Hi. I am currently facing the same issue. I have checked everything. All my labels and images have 0 and 1 values. I am still getting this error

EDIT:

Here's what helped me:

`
nnUNetv2_plan_and_preprocess -d 509 --verify_dataset_integrity -c 3d_fullres --verbose
`
It helped me vizualize the bad images. So it might help that in preprocessing, it should automatically skip bad examples. Is there a way to do that?
